File: lib/mutations/UncancelTweet.ts
import { graphql, Environment, commitMutation } from "react-relay"
import { UncancelTweetMutationResponse } from "../__generated__/UncancelTweetMutation.graphql"

const mutation = graphql`
  mutation UncancelTweetMutation($input: UncancelTweetInput!) {
    uncancelTweet(input: $input) {
      tweet {
        id
        status
      }
    }
  }
`

function getOptimisticResponse(id: string): UncancelTweetMutationResponse {
  return {
    uncancelTweet: {
      tweet: {
        id,
        status: "DRAFT",
      },
    },
  }
}

export function uncancelTweet(environment: Environment, id: string) {
  const variables = { input: { id } }

  commitMutation(environment, {
    mutation,
    variables,
    optimisticResponse: getOptimisticResponse(id),
    // TODO add update config to move this to the other range
  })
}

File: ios/Courier/CourierNext/Mutations/UncancelTweet.swift
import Relay
import RelaySwiftUI

private let mutation = graphql("""
mutation UncancelTweetMutation($input: UncancelTweetInput!) {
  uncancelTweet(input: $input) {
    tweetGroup {
      id
      status
    }
  }
}
""")

extension Mutation.Mutator where Operation == UncancelTweetMutation {
    func commit(id: String) {
        commit(
            variables: .init(input: UncancelTweetInput(id: id)),
            optimisticResponse: [
                "uncancelTweet": [
                    "tweetGroup": [
                        "id": id,
                        "status": "DRAFT",
                    ]
                ]
            ]
        )
    }
}
